### DISCUSSION TITLE
Joins in one-many with not equal criteria

### DISCUSSION THREAD

**Discussion by rmrbytes** [supabase/supabase#11518] [Questions]
Hello:

I have a **content** table  `{id, contentcode, contentinfo }` and a user **subscription** table which stores all content subscribed by the users `(id, contentcode}`. **contentcode** is foreign keys 

I wish to know **all contents that are not part** of the subscription table

Before filtering I tried to first retrieve:

```
 await supabase
  from('content')
  select('*, subscription(contentcode)')
```

This retrieves all records with the join but it gives the subscription object **as an array** in the result. 

Is that how it works? Is there a way to get the joined table as a column and not an array?

Next, I would like to apply the filter to remove all records that are in subscription table. I tried

```
 await supabase
  from('content')
  select('*, subscription(contentcode)')
  .eq('subscription.content', null)
```

Thanks

**Comment by GaryAustin1** [comment]
You probably will need the new PostgREST 11 for this.
https://supabase.com/blog/postgrest-11-prerelease
See anti-joins about 1/2 down.

Also note you cannot use .eq() with null.  You need to use .is(), but that is not your problem here.

I have no idea when 11 is available generally.

**Reply by rmrbytes** [reply]
Thanks @GaryAustin1. I will look up the link. 

But before applying the where I am struggling with the array result of my joined table. I am wondering if I could apply an `.eq` or `neq` condition based on the array length.

**Reply by GaryAustin1** [reply]
You are getting to the point I usually just bail and use an rpc and do the SQL....

This might be related:  https://github.com/supabase/supabase/discussions/1113#discussioncomment-594673  but I don't know if that translates to a filter condition.

These discussion have a dozen or so related issues if you can get a hit with search terms.  
Ultimately everything comes from this https://postgrest.org/en/stable/api.html as far as what you can do.

**Comment by rmrbytes** [comment]
Yup. you are right. Will give it a try and share back. Thanks @GaryAustin1
